commands/hooks/yacc.go

```diff
@@ -296,3 +307,157 @@ func (command *YaccHookDisableCommand) DisableAction(context *cli.Context) error
 
 	return nil
 }
+
+// YaccHookSettingsCommand define the command to get setting for YACC hook
+type YaccHookSettingsCommand struct {
+	Settings *settings.BitAdminSettings
+	flags    *YaccHookGetSettingsCommandFlags
+}
+
+// YaccHookGetSettingsCommandFlags define the flags of the YaccHookSettingsCommand
+type YaccHookGetSettingsCommandFlags struct {
+	project    string
+	repository string
+}
+
+// GetCommand provide a ready to use cli.Command
+func (command *YaccHookSettingsCommand) GetCommand() cli.Command {
+	return cli.Command{
+		Name:   "get-settings",
+		Usage:  "Get settings for Yet Another Commit Checker hook",
+		Action: command.GetSettings,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:        "project",
+				Usage:       "The `<project_key>` where the repository will be created",
+				Destination: &command.flags.project,
+			},
+			cli.StringFlag{
+				Name:        "repository",
+				Usage:       "The `<repository_name>` to create",
+				Destination: &command.flags.repository,
+			},
+		},
+		BashComplete: func(c *cli.Context) {
+			helper.AutoComplete(c, command.Settings.GetFileCache())
+		},
+	}
+}
+
+// GetSettings contains logic to get current yacc settings
+func (command *YaccHookSettingsCommand) GetSettings(context *cli.Context) error {
+	client, err := command.Settings.GetAPIClient()
+	if err != nil {
+		return err
+	}
+
+	if len(command.flags.project) <= 0 {
+		return errors.New("--project flag is required")
+	}
+	if len(command.flags.repository) <= 0 {
+		return errors.New("--repository flag is required")
+	}
+
+
+	yacc, err := GetHookSettings(
+		command.flags.project,
+		command.flags.repository,
+		yaccHookKey,
+		client,
+	)
+	if err != nil {
+		return err
+	}
+
+	data, err := json.Marshal(yacc)
+	if err != nil {
+		return err
+	}
+
+	isCorrect  := "NO"
+	isOK, diff := isYACCSettingsCorrectWithDiff(yacc)
+
+	if isOK == true {
+		isCorrect = "YES"
+	}
+
+	fmt.Printf("%s# %s# %s", isCorrect, data, diff.String())
+
+	return nil
+}
+
+// GetHookSettings return the YACC hook settings from bitbucket
+func GetHookSettings(projectKey, repositorySlug string, hookKey string, bc *bitclient.BitClient) (YaccSettings, error) {
+	response := YaccSettings{}
+
+	_, err := bc.DoGet(
+		fmt.Sprintf("/projects/%s/repos/%s/settings/hooks/%s/settings", projectKey, repositorySlug, hookKey),
+		nil,
+		&response,
+	)
+
+	return response, err
+}
+
+// isYACCSettingsCorrectWithDiff Compares YACC hook settings for a repository and check whether these are matching the desired default settings
+func isYACCSettingsCorrectWithDiff(yaccSettings YaccSettings) (bool, DiffReporter) {
+	defaultYACCSettings := YaccSettings {
```

Comment:
Thoses settings are user specific, not sure it has to be part of the repository. Maybe exposing it as flags would make it more generic ?

Reply:
I split this to 2 command. "get-settings", "diff-settings". Think is better like this

Reply:
Yup perfect. Just restore bitclient in the Gopck.toml back to master, and good to go ;)

Reply:
done